testutil/garbage/garbage.go

```diff
@@ -0,0 +1,304 @@
+	options := applyOptions(opts...)
+	rand.Seed(time.Now().UnixNano())
+	_, n := generate(options.blockSize, options)
+	return n
+}
+
+func generate(count uint64, opts Options) (uint64, datamodel.Node) {
+	weights := opts.weights
+	if opts.initialWeights != nil {
+		weights = opts.initialWeights
+		opts = Options{weights: opts.weights}
+	}
+	totWeight := 0
+	for _, kind := range kinds {
+		totWeight += weights[kind]
+	}
+	r := rand.Float64() * float64(totWeight)
+	var wacc int
+	for _, kind := range kinds {
+		wacc += weights[kind]
+		if float64(wacc) >= r {
+			return generators[kind](count, opts)
+		}
+	}
+	panic("bad options")
+}
+
+func rndSize(bias uint64) uint64 {
+	// @rv: OK, I'll admit that I've completely forgotten how I arrived at this, but it
+	// generates numbers that bias toward `bias`, but not strictly, they may vary wildly
+	// but be drawn toward the lower end of the range and tend to be suitable for lengths
+	// of arrays and such
+	m := math.Abs(math.Floor(5/(1-(math.Pow(rand.Float64(), 2))) - 5 + rand.Float64()*float64(bias)))
+	return uint64(m)
+}
+
+func rndRune() rune {
+	return runes[rand.Intn(len(runes))]
+}
+
+func listGenerator(count uint64, opts Options) (uint64, datamodel.Node) {
+	len := rndSize(5)
+	lb := basicnode.Prototype.List.NewBuilder()
+	la, err := lb.BeginList(int64(len))
+	if err != nil {
+		panic(err)
+	}
+	size := uint64(0)
+	for i := uint64(0); i < len && size < count; i++ {
+		c, n := generate(count-size, opts)
+		la.AssembleValue().AssignNode(n)
+		size += c
+	}
+	err = la.Finish()
+	if err != nil {
+		panic(err)
+	}
+	return size, lb.Build()
+}
+
+func mapGenerator(count uint64, opts Options) (uint64, datamodel.Node) {
+	length := rndSize(5)
+	mb := basicnode.Prototype.Map.NewBuilder()
+	ma, err := mb.BeginMap(int64(length))
+	if err != nil {
+		panic(err)
+	}
+	size := uint64(0)
+	keys := make(map[string]struct{})
+	for i := uint64(0); i < length && size < count; i++ {
+		var key string
+		for {
+			c, k := stringGenerator(5, opts)
+			key = must.String(k)
+			if _, ok := keys[key]; !ok && len(key) > 0 {
+				keys[key] = struct{}{}
+				size += c
+				break
+			}
+		}
+		sz := count - size
+		if size >= count { // the case where we've blown our budget already on the key
+			sz = 5
+		}
+		c, value := generate(sz, opts)
+		size += c
+		err := ma.AssembleKey().AssignString(key)
+		if err != nil {
+			panic(err)
+		}
+		err = ma.AssembleValue().AssignNode(value)
+		if err != nil {
+			panic(err)
+		}
+	}
+	err = ma.Finish()
+	if err != nil {
+		panic(err)
+	}
+	return size, mb.Build()
+}
+
+func stringGenerator(count uint64, opts Options) (uint64, datamodel.Node) {
+	len := rndSize(count)
+	sb := strings.Builder{}
+	for i := uint64(0); i < len; i++ {
+		sb.WriteRune(rndRune())
+	}
+	return len, basicnode.NewString(sb.String())
+}
+
+func bytesGenerator(count uint64, opts Options) (uint64, datamodel.Node) {
+	len := rndSize(count)
+	ba := make([]byte, len)
+	rand.Read(ba)
```

Comment:
nit: you should check for an error here, too. even math/rand can fail, for instance if a custom source is provided.